ledger/participant-integration-api/src/test/suite/scala/platform/index/ContractStoreBasedMaximumLedgerTimeServiceSpec.scala

```diff
@@ -0,0 +1,256 @@
+        contractId3,
+        contractId4,
+      )
+    ).map(
+      _ shouldBe Max(timestamp1)
+    )
+  }
+
+  it should "find no maximum ledger time if ids is empty" in {
+    testeeWithFixture(
+    ).lookupMaximumLedgerTimeAfterInterpretation(
+      Set(
+      )
+    ).map(
+      _ shouldBe NotAvailable
+    )
+  }
+
+  it should "find the maximum ledger time if there for only one active contract" in {
+    testeeWithFixture(
+      contractId1 -> active(timestamp1)
+    ).lookupMaximumLedgerTimeAfterInterpretation(
+      Set(
+        contractId1
+      )
+    ).map(
+      _ shouldBe Max(timestamp1)
+    )
+  }
+
+  it should "find the maximum ledger time if there are some contracts which cannot be found" in {
+    testeeWithFixture(
+      contractId1 -> active(timestamp1),
+      contractId2 -> NotFound,
+      contractId3 -> active(timestamp3),
+      contractId4 -> NotFound,
+    ).lookupMaximumLedgerTimeAfterInterpretation(
+      Set(
+        contractId1,
+        contractId2,
+        contractId3,
+        contractId4,
+      )
+    ).map(
+      _ shouldBe Max(timestamp3)
+    )
+  }
+
+  it should "find no maximum ledger time if none of the contracts can be found" in {
+    testeeWithFixture(
+      contractId1 -> NotFound,
+      contractId2 -> NotFound,
+      contractId3 -> NotFound,
+      contractId4 -> NotFound,
+    ).lookupMaximumLedgerTimeAfterInterpretation(
+      Set(
+        contractId1,
+        contractId2,
+        contractId3,
+        contractId4,
+      )
+    ).map(
+      _ shouldBe NotAvailable
+    )
+  }
+
+  it should "find no maximum ledger time if for the one contract cannot be found" in {
+    testeeWithFixture(
+      contractId1 -> NotFound
+    ).lookupMaximumLedgerTimeAfterInterpretation(
+      Set(
+        contractId1
+      )
+    ).map(
+      _ shouldBe NotAvailable
+    )
+  }
+
+  it should "return the archived contract, if one of the contracts is archived" in {
+    testeeWithFixture(
+      contractId1 -> active(timestamp1),
+      contractId2 -> active(timestamp2),
+      contractId3 -> ContractState.Archived,
+      contractId4 -> active(timestamp4),
+    ).lookupMaximumLedgerTimeAfterInterpretation(
+      Set(
+        contractId1,
+        contractId2,
+        contractId3,
+        contractId4,
+      )
+    ).map(
+      _ shouldBe MaximumLedgerTime.Archived(Set(contractId3))
+    )
+  }
+
+  it should "return one of the archived contracts, if two of the contracts are archived" in {
+    testeeWithFixture(
+      contractId1 -> ContractState.Archived,
+      contractId2 -> active(timestamp2),
+      contractId3 -> ContractState.Archived,
+      contractId4 -> active(timestamp4),
+    ).lookupMaximumLedgerTimeAfterInterpretation(
+      Set(
+        contractId1,
+        contractId2,
+        contractId3,
+        contractId4,
+      )
+    ).map { result =>
+      val MaximumLedgerTime.Archived(archivedResults) = result
+      archivedResults.size shouldBe 1
```

Comment:
Implementation detail or test requirement ?

Reply:
well it depends how you look at it:
* from the interface point of view not that important
* in the implementation we always return one by design (failing fast)

I was aiming for more of an implementation focused white-box test here, sine this is really a unit test of the implementation.